frontend/src/components/Header/components/CuratorAPIKeyGenerator/index.tsx

```diff
@@ -0,0 +1,80 @@
+import { Dialog, Intent, MenuItem } from "@blueprintjs/core";
+import {
+  Button as SDSButton,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+} from "czifui";
+import { useCallback, useRef, useState } from "react";
+import {
+  APIKeyResponse,
+  generateCuratorAuthKey,
+  useCuratorAuthKeyID,
+} from "src/common/queries/curation";
+import Toast from "src/views/Collection/components/Toast";
+import { APIDisclaimerP, FullWidthCallout, StyledInputGroup } from "./style";
+
+export default function CuratorAPIKeyGenerator(): JSX.Element {
+  const { data: apiKeyID } = useCuratorAuthKeyID();
+
+  const apiKeyResponse = useRef({ id: apiKeyID ?? "" } as APIKeyResponse);
+
+  const [isOpen, setIsOpen] = useState(false);
+  const handleOpen = useCallback(async () => {
+    try {
+      const data = await generateCuratorAuthKey();
+      if (!data?.key) throw new Error("No key returned");
+      apiKeyResponse.current = data;
+      setIsOpen(!isOpen);
+    } catch (error) {
+      Toast.show({
+        intent: Intent.DANGER,
+        message: "An error occurred while generating an API Key",
+      });
+    }
+  }, [isOpen]);
+  const handleClose = useCallback(() => setIsOpen(false), []);
```

Comment:
bonus: should we not close the modal when close reason is `backdropClick`? Just thinking that users might accidentally close the modal 😆

Reply:
Added in [this PR](https://github.com/chanzuckerberg/single-cell-data-portal/pull/2037) in case needed